src/OpenRiaServices.Server.EntityFrameworkCore/Framework/EFCoreTypeDescriptor.cs

```diff
@@ -0,0 +1,251 @@
+
+                // All members marked with TimestampAttribute (inferred or explicit) need to
+                // have [Editable(false)] and [RoundtripOriginal] applied
+                if (hasTimestampAttribute)
+                {
+                    inferRoundtripOriginalAttribute = true;
+
+                    if (editableAttribute == null)
+                    {
+                        editableAttribute = new EditableAttribute(false);
+                    }
+                }
+
+                // Add RTO to this member if required. If this type has a timestamp
+                // member that member should be the ONLY member we apply RTO to.
+                // Dont apply RTO if it is an association member.
+                bool isForeignKeyMember = _foreignKeyMembers != null && _foreignKeyMembers.ContainsKey(property.Name);
+                if ((_timestampProperty == null || _timestampProperty == property) &&
+                    (inferRoundtripOriginalAttribute || isForeignKeyMember) &&
+                    pd.Attributes[typeof(AssociationAttribute)] == null)
+                {
+                    if (pd.Attributes[typeof(RoundtripOriginalAttribute)] == null)
+                    {
+                        attributes.Add(new RoundtripOriginalAttribute());
+                    }
+                }
+            }
+
+            // Add the Editable attribute if required
+            if (editableAttribute != null && pd.Attributes[typeof(EditableAttribute)] == null)
+            {
+                attributes.Add(editableAttribute);
+            }
+
+            // Add AssociationAttribute if required for the specified property
+            if (_entityType.FindNavigation(pd.Name) is INavigation navigation)
+            {
+                bool isManyToMany = navigation.IsCollection() && navigation.FindInverse()?.IsCollection() == true;
+                if (!isManyToMany)
+                {
+                    var assocAttrib = (AssociationAttribute)pd.Attributes[typeof(AssociationAttribute)];
+                    if (assocAttrib == null)
+                    {
+                        assocAttrib = TypeDescriptionContext.CreateAssociationAttribute(navigation);
+                        attributes.Add(assocAttrib);
+                    }
+                }
+
+            }
+
+            return attributes.ToArray();
+        }
+
+        /// <summary>
+        /// Determines whether the specified property is an Entity member that
+        /// should be excluded.
+        /// </summary>
+        /// <param name="pd">The property to check.</param>
+        /// <returns>True if the property should be excluded, false otherwise.</returns>
+        internal static bool ShouldExcludeEntityMember(PropertyDescriptor pd)
+        {
+            //TODO: remove this, EntityState is not part of entities 
+
+            // exclude EntityState members
+            if (pd.PropertyType == typeof(EntityState)) // TODO: Maybe also check pd.Component type
```

Comment:
Should pd.ComponentType be checked?

Reply:
Or should all of this be removed, the comment above suggests that entitystate is not part of enties

Reply:
I think the comment on row 239 can be saftly removed, or merged with 241 to clarify that it is `IChangeDetector.EntityState` we want to exclude.
 EF core does support an EntityState property when implementing the `IChangeDetector` interface and it should not be considered part of the model / datacontract.


```suggestion
            if (pd.PropertyType == typeof(EntityState))
```

Reply:
Looks good!